### PR TITLE
Adds jenkins_protocol_https variable

### PR DESCRIPTION
**Pull request by pomegranited**
When 'jenkins_protocol_https: true' (default value), this configures nginx with `proxy_redirect` from `http:` to `https:`.  Ref [edx-analytics-configuration jenkins.j2](https://github.com/edx/edx-analytics-configuration/blob/master/roles/jenkins/templates/etc/nginx/sites-available/jenkins.j2#L8-L11).

This change maintains the default behaviour, but makes the `proxy_redirect` config optional.

**JIRA tickets**: Discovered while deploying Jenkins to an OpenCraft client site ([OC-1561](https://tasks.opencraft.com/browse/OC-1561)).

**Related PRs**: 
#3026 , #3027

**Testing instructions**:

Build a `analytics-jenkins` vagrant instance: 

``` bash
git clone https://github.com/open-craft/configuration
cd configuration
git checkout jill/jenkins_master_protocol
mkvirtualenv analytics-jenkins
make requirements  # requires libmysqlclient-dev
vim ansible.yml   # see below
export VAGRANT_JENKINS_LOCAL_VARS_FILE=/path/to/ansible.yml
cd vagrant/base/analytics_jenkins
vim Vagrantfile  # see below

# should skip the final task that creates and runs AnalyticsSeedJob, but if it doesn't, don't worry about that final failure :)
ANSIBLE_ARGS='--skip-tags=jenkins-seed-job' vagrant up 
```

`ansible.yml`:

``` yaml

---
JENKINS_ANALYTICS_AUTH_REALM: unix
JENKINS_ANALYTICS_USER_PASSWORD_PLAIN: password
jenkins_protocol_https: false
jenkins_server_name: localhost
```

Add to `vagrant/base/analytics_jenkins/Vagrantfile`:

```
...
  unless ENV['VAGRANT_NO_PORTS']
    config.vm.network :forwarded_port, guest: 8080, host: 8080  # Jenkins
    config.vm.network :forwarded_port, guest: 80, host: 80  # Jenkins proxied by nginx
  end
...
```
1. Visit http://localhost
2. Login as jenkins user
3. Note that you're not redirected to https.

To undo this change and watch it break:
1. Add `proxy_redirect http: https:` to the vagrant instance's `/etc/nginx/sites-enabled/jenkins`
2. `sudo service nginx restart`
3. Logout of jenkins, and try to log back in.  See it redirect to https.

**Reviewers**
- [x] @jbzdak  
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fconfiguration&number=3025

File: playbooks/roles/jenkins_master/defaults/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ jenkins_group: "edx"
 jenkins_server_name: "jenkins.testeng.edx.org"
 jenkins_port: 8080
 jenkins_nginx_port: 80
+jenkins_protocol: https
```
Comment (jbzdak) [inline playbooks/roles/jenkins_master/defaults/main.yml]: This variable works as a boolean switch, so I'd rather have this explicit, and rename it like: `jenkins_nginx_rewrite_to_https: true`, or something similar.

Comment (pomegranited) [inline playbooks/roles/jenkins_master/defaults/main.yml]: Yep, that's nicer.  Fixed in [a31d8f2](https://github.com/edx/configuration/pull/3025/commits/a31d8f20aacb3c7fa54d129f20ab30aafd222f44).

File: playbooks/roles/jenkins_master/defaults/main.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ jenkins_group: "edx"
 jenkins_server_name: "jenkins.testeng.edx.org"
 jenkins_port: 8080
 jenkins_nginx_port: 80
+jenkins_protocol_https: true
```
Comment (jbzdak) [inline playbooks/roles/jenkins_master/defaults/main.yml:7]: NIT: instructions not updated to reflect new (boolean) variable

Comment (pomegranited) [inline playbooks/roles/jenkins_master/defaults/main.yml:7]: Oops, sorry!

**Comment by jbzdak** [comment]
:+1: Works as advertised.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! I've created [OSPR-1261](https://openedx.atlassian.net/browse/OSPR-1261) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by nedbat** [comment]
@edx/devops Can we start this review?

**Comment by feanil** [comment]
:+1:   @edx/devops second review.

**Comment by pomegranited** [comment]
@nedbat Does @feanil's review mean this is ready to merge?  Thank you both for getting this through!

**Comment by nedbat** [comment]
@pomegranited Feanil is tagging another devops engineer to get another thumb.  It won't take long :)

File: playbooks/roles/jenkins_master/defaults/main.yml (hunk 2 above)
Comment (jibsheet) [inline playbooks/roles/jenkins_master/defaults/main.yml:7]: Did this doc get updated?

**Comment by jibsheet** [comment]
👍 but see question about doc.

File: playbooks/roles/jenkins_master/defaults/main.yml (hunk 2 above)
Comment (pomegranited) [inline playbooks/roles/jenkins_master/defaults/main.yml:7]: @jibsheet He was referring to the PR instructions, and they are updated.

Comment (jibsheet) [inline playbooks/roles/jenkins_master/defaults/main.yml:7]: Great, you should go ahead and merge this and the other 2.

**Pull request merged**